semver/comparator_format.ts

```diff
@@ -12,5 +12,5 @@ import { format } from "./format.ts";
  */
 export function comparatorFormat(comparator: Comparator): string {
   const { semver, operator } = comparator;
-  return `${operator}${format(semver)}`;
+  return `${operator}${format(semver ?? comparator)}`;
```

Comment:
Nice. Can we please have added functionality? We want to make sure that both `SemVer` and `Comparator` inputs work fine on all these functions, including the others in this PR.

Reply:
How do you mean exactly?

Reply:
I added some test cases. PTAL.